lib/codecs/src/decoding/format/avro.rs

```diff
@@ -0,0 +1,366 @@
+    ) -> vector_common::Result<SmallVec<[Event; 1]>> {
+        // Avro has a `null` type which indicates no value.
+        if bytes.is_empty() {
+            return Ok(smallvec![]);
+        }
+
+        let bytes = if self.strip_schema_id_prefix {
+            if bytes.len() >= CONFLUENT_SCHEMA_PREFIX_LEN && bytes[0] == CONFLUENT_MAGIC_BYTE {
+                bytes.slice(CONFLUENT_SCHEMA_PREFIX_LEN..)
+            } else {
+                return Err(vector_common::Error::from(
+                    "Expected avro datum to be prefixed with schema id",
+                ));
+            }
+        } else {
+            bytes
+        };
+
+        let value = apache_avro::from_avro_datum(&self.schema, &mut bytes.reader(), None)?;
+
+        let apache_avro::types::Value::Record(fields) = value else {
+            return Err(vector_common::Error::from("Expected an avro Record"));
+        };
+
+        let mut log = LogEvent::default();
+        for (k, v) in fields {
+            log.insert(event_path!(k.as_str()), try_from(v)?);
+        }
+
+        let mut event = Event::Log(log);
+        let event = match log_namespace {
+            LogNamespace::Vector => event,
+            LogNamespace::Legacy => {
+                if let Some(timestamp_key) = log_schema().timestamp_key_target_path() {
+                    let log = event.as_mut_log();
+                    if !log.contains(timestamp_key) {
+                        let timestamp = Utc::now();
+                        log.insert(timestamp_key, timestamp);
+                    }
+                }
+                event
+            }
+        };
+        Ok(smallvec![event])
+    }
+}
+
+// Can't use std::convert::TryFrom because of orphan rules
+pub fn try_from(value: AvroValue) -> vector_common::Result<VrlValue> {
+    // Very similar to avro to json see `impl std::convert::TryFrom<AvroValue> for serde_json::Value`
+    // LogEvent has native support for bytes, so it is used for Bytes and Fixed
+    match value {
+        AvroValue::Array(array) => {
+            let mut vector = Vec::new();
+            for item in array {
+                vector.push(try_from(item)?);
+            }
+            Ok(VrlValue::Array(vector))
+        }
+        AvroValue::Boolean(boolean) => Ok(VrlValue::from(boolean)),
+        AvroValue::Bytes(bytes) => Ok(VrlValue::from(bytes)),
+        AvroValue::Date(d) => Ok(VrlValue::from(d)),
+        AvroValue::Decimal(ref d) => Ok(<Vec<u8>>::try_from(d)
+            .map(|vec| VrlValue::Array(vec.into_iter().map(VrlValue::from).collect()))?),
+        AvroValue::Double(double) => Ok(VrlValue::from_f64_or_zero(double)),
+        AvroValue::Duration(d) => Ok(VrlValue::Array(
+            <[u8; 12]>::from(d)
+                .into_iter()
+                .map(VrlValue::from)
+                .collect(),
+        )),
+        AvroValue::Enum(_, string) => Ok(VrlValue::from(string)),
+        AvroValue::Fixed(_, bytes) => Ok(VrlValue::from(bytes)),
```

Comment:
To workaround the roundtrip error you can do: 
`Ok(VrlValue::from(String::from_utf8_lossy(&bytes))),`

And then remove `fixed.avro` from the exclusion list.

Reply:
The [avro fixed](https://avro.apache.org/docs/1.11.1/specification/#fixed) is (usize, Vec<u8>) not String, converting it to String may have some side effect which `String::from_utf8_lossy` will replace some bytes.

[avro resolve_fixed](https://github.com/apache/avro/blob/bfc19adae5a55e119dc5ff3c1d9f8455e11557f3/lang/rust/avro/src/types.rs#L893-L905C4)